### DISCUSSION TITLE
Windows Auto Enrollement

### DISCUSSION THREAD

**Discussion by habsfanongit** [smallstep/certificates#1107] [Q&A]
Hello,
Does SmallStep Enterprise or Open Source edition support windows auto enrollment in an AD environment? 

Thanks

**Comment by maraino** [comment]
Are you referring to this ["Set up enrollment for Windows devices"](https://learn.microsoft.com/en-us/mem/intune/enrollment/windows-enroll#enable-windows-automatic-enrollment)

This talks about configuring an MDM enrollment, this is not supported, but it can be supported if the MDM can support [ACME device attestation draft](https://datatracker.ietf.org/doc/html/draft-bweeks-acme-device-attest-01), but they won't do that yet.

But you can use ActiveDirectory OAuth/OpenID implementation to get users' certificates after signing in. Doing it automatically might be possible, but that process should be implemented somehow in Windows.
